source/extensions/filters/listener/tls_inspector/tls_inspector.cc

```diff
@@ -228,14 +228,20 @@ std::vector<absl::string_view> Filter::getAlpnProtocols(const unsigned char* dat
     return protocols;
   }
 
-  absl::string_view str(reinterpret_cast<const char*>(data));
   for (unsigned int i = 0; i < len;) {
-    uint32_t protocol_length = 0;
-    protocol_length <<= 8;
-    protocol_length |= data[i];
-    ++i;
-    absl::string_view protocol(str.substr(i, protocol_length));
-    protocols.push_back(protocol);
+    uint8_t protocol_length = data[i++];
+
+    if (protocol_length == 0) {
+      ENVOY_LOG(debug, "tls inspector: ALPN protocol length is zero");
```

Comment:
I wonder if emitting a warning in combination with a switch to `ENVOY_LOG_PERIODIC_TO_LOGGER` [1] would be better here. wdyt? 

[1] https://github.com/envoyproxy/envoy/blob/e727e1f50b33730a3a4fc061f95bcdd6d0c48768/source/common/common/logger.h#L541

Reply:
I wasn't too sure about logging level to use. I had a look around and it seemed to me like warning/error log messages are generally used to report on the (mal)functioning of envoy itself, rather than to report on erroneous content flowing through envoy, but I'll change it as suggested.

Reply:
The thing is that with regular logging this could result in far too much log spew. But i think that hitting these edge cases is worth surfacing to the user. With the periodic log function, we don't be flooding the logs while still surfacing that this stuff is happening. That was my reasoning.

Reply:
Done